bgpd/bgp_open.c

```diff
@@ -543,6 +543,25 @@ static as_t bgp_capability_as4(struct peer *peer, struct capability_header *hdr)
 	return as4;
 }
 
+static int bgp_capability_ext_message(struct peer *peer,
+				      struct capability_header *hdr)
+{
+	SET_FLAG(peer->cap, PEER_CAP_EXT_MESSAGE_RCV);
+
+	if (CHECK_FLAG(peer->cap, PEER_CAP_EXT_MESSAGE_ADV))
+		peer->max_packet_size = BGP_MAX_EXT_MESSAGE_PACKET_SIZE;
```

Comment:
Can we check if we are capable only then set the buffer if we receive the capability?

From RFC 8654

5.  Error Handling

   A BGP speaker that has the ability to use BGP Extended Messages but
   has not advertised the BGP Extended Message Capability, presumably
   due to configuration, MUST NOT accept a BGP Extended Message.  A
   speaker MUST NOT implement a more liberal policy accepting BGP
   Extended Messages.

Reply:
Actually this is set in a function which already handles this capability. Or I don’t understand what do you mean.

Reply:
I was thinking that we should drop the capability if we have not advertised as per the section in RFC. This means not setting the flag SET_FLAG(peer->cap, PEER_CAP_EXT_MESSAGE_RCV); but then when I see the code it does look like you are using this flag only for display and no decisions are taken here.